src/callstack.cpp

```diff
@@ -60,9 +60,98 @@ static const char* alloc_string( callstack_string_buffer_t* buf, const char* str
 	#include <execinfo.h>
 	#include <stdio.h>
 	#include <stdlib.h>
+	#include <string.h>
 	#include <unistd.h>
 	#include <cxxabi.h>
 
+	typedef struct mmap_entry_t {
+		struct mmap_entry_t *next;
+
+		unsigned long range_start;
+		unsigned long range_end;
+		char *perms;
+		unsigned long file_offset;
+		char *dev;
+		char *inode;
+		// can be null if there is no path specified
+		char *path;
+	} mmap_entry_t;
+
+	static mmap_entry_t *parse_mmaps(const char *maps_file){
+		FILE* maps = fopen(maps_file, "r");
+		if(!maps){
+			perror("fopen");
+			fprintf(stderr, "failed to open %s\n", maps_file);
+			return NULL;
+		}
+
+		mmap_entry_t *first = NULL;
+		mmap_entry_t *cur = NULL;
+
+		while(!feof(maps)){
+			char buf[4096];
+			char *ret = fgets(buf, 4096, maps);
+			if(!ret)
+				break;
+
+			mmap_entry_t *next = (mmap_entry_t*) malloc(sizeof(mmap_entry_t));
+			next->next = NULL;
+
+			// from man `proc`:
+			// address perms offset dev inode pathname
+			sscanf(buf, "%lx-%lx %ms %lx %ms %ms %ms\n",
```

Comment:
there is really no need for this lib to dynamically alloc all the strings as you don't use them, so I would just change the %ms to %*s and remove the string-members from mmap_entry_t.

Reply:
Fixed. I copied this from something else I wrote that did use (some of) them